### DISCUSSION TITLE
Please consider removing the banner in future versions

### DISCUSSION THREAD

**Discussion by daniel-brenot** [pixijs/pixijs#8705] [General]
I just wanted to start off by saying that I absolutely love PIXI.js. It has been great in my most recent project given its stellar performance and simple api, and I will definitely be giving a shoutout in my README file.

Something I noticed right away when I downloaded the library is that it automatically prints to the console log to advertise itself. While I understand the reason behind this is for publicity, I think it was a mistake. The banner is caught by integration tests and by automatic logging software.

Yes it can be turned off, but this can be annoying to have to configure for all of your tests, especially if they have different bases for example. I have an otherwise clean main.js file for my angular project that has the line of code for pixi be one of the first things that is called. While this isn't a major intrusion at the moment, I can't imagine many people would be thrilled if all the libraries that they used did this(imagine if lodash advertised itself).

What's worse is the response to people asking about disabling this for legitimate reasons. I've seen people be called a "jerkface" for asking to disable this, and being told that they will look down their nose at them. They are called lazy and insulted at any mention of this being an annoyance. Maybe this was meant as a joke, but it's not terribly funny to ostracize part of the community using the library. It's not personal that developers don't want to see this banner.

Additionally, if someone wants to know what rendering library is being used on a page, they can use wappalyser to find out(this is why I actually first found out about pixi!)

I appreciate any response and I hope my grievances are understood as they are meant. This isn't meant as an attack against the maintainers or contributors, but as a request that I think will genuinely make the library better.

**Comment by bigtimebuddy** [comment]
I converted to a discussion since it’s a little more helpful for debating and having threads.

@daniel-brenot Thank you for your thoughtful feedback about the console banner.

As one of the lead maintainers of the project, I largely agree with the points here. Pixi is an outlier for advertising itself in this way among web projects. I remember a similar issue a few years ago around projects asking for contributions in postinstall npm scripts. Largely the node/JS community came to the consensus that this wasn’t acceptable and funding information can now be added to the package.json.

This project was first incubated by the design agency GoodBoy Digital, and much of the public presentation of Pixi was created in a way to promote their expertise in the space. But Pixi is no longer financially dependent or managed by GBD (though Mat and Sean are still active on the project), so there’s less of a need to use Pixi as a vehicle for attracting other contracts or work.

That said, there are a handful of unsavory industries that are huge consumers of Pixi. Generally these industries are not big contributors (financially or materially) to the project although sometimes individual developers are active. As free licensed software, there is not a lot we can do about this dynamic. The banner and public pressure to not remove it has, I believe, been a way to make sure folks at least give back in the form of advertising. I’m not saying this is justified, but it’s been a historical element.

The name calling and pressure campaign around this issue is unacceptable and since many of the posts referenced here, the project has added contributing guidelines and the core team has had significant turnover.

Lastly, while the banner is kinda loud and colorful, there is one very practical thing about it: lets us know if the rendering context is Canvas, WebGL or WebGL 2.  This is helpful debugging with users and in some environments. There are other ways to do this, such as an opt-in flag. If we remove the banner, I’d like some opt-in debugging info to replace it.
